drivers/accel/adxl367/adxl367.c

```diff
@@ -511,14 +515,23 @@ int adxl367_set_offset(struct adxl367_dev *dev, uint16_t x_offset,
 				    ADXL367_XYZ_AXIS_OFFSET_MASK);
 	if (ret)
 		return ret;
+	dev->x_offset = x_offset;
```

Comment:
updating of this field should happen after the reg write, only if reg write succeeded (same for y, z)